### PR TITLE
Update 2.1 release notes.

### PR DESCRIPTION
**Pull request by monopole**

### CODE REVIEW

File: docs/v_2.1.0.md [hunk 1]
```diff
@@ -1,25 +1,179 @@
...
+  (`generators` and `tranformers`) and one existing
+  field (`resources`) now accept arguments
+  that were once accepted only by `bases`.
+
+  The fact that the `generators` and
+  `transformers` field accept [bases]
+  (i.e. kustomization directories) and the fact
+  that generator and transformer configuration
+  objects are normal k8s resources means that one
+  can generate or transform a generator or a
```
Comment (Liujingfang1) [inline docs/v_2.1.0.md]:
```
generate or transform a generator or a transformer
```
How about
```
customize a generator or a transformer
```

File: docs/v_2.1.0.md [hunk 2]
```diff
@@ -1,25 +1,179 @@
...
+a package structure intentially exported as one or more
+[Go modules] for use in other programs
+(kubectl, kubebuilder, etc.) and in kustomize
+plugins (see below).
 
- * Last release with completely unsupported access to `pkg`.
+
+## Inventory generation for pruning
+
+![pruning dead branches][imgPruning]
```
[the remark this answers is not in the document]
Comment (monopole) [reply docs/v_2.1.0.md]: That's markdown notation for https://www.w3schools.com/tags/att_img_alt.asp

File: docs/v_2.1.0.md [hunk 3]
```diff
@@ -1,25 +1,179 @@
...
 
-   * New `envs` sub-field introduced to both
-     `configMapGenerators` and `secretGenerators`,
-     replacing the deprecated `env` field
-     (allowing mixins of k=v pairs from different files).
+![gopher with boxes][imgModules]
 
- * GO modules
+Kustomize now defines it dependencies in a top
+level `go.mod` file.  This the first step towards
```
[the remark this answers is not in the document]
Comment (monopole) [reply docs/v_2.1.0.md]: lol, thanks.

File: docs/v_2.1.0.md [hunk 4]
```diff
@@ -1,25 +1,179 @@
...
+## Go modules
 
-   * New `envs` sub-field introduced to both
-     `configMapGenerators` and `secretGenerators`,
-     replacing the deprecated `env` field
-     (allowing mixins of k=v pairs from different files).
+![gopher with boxes][imgModules]
 
- * GO modules
+Kustomize now defines it dependencies in a top
```
[the remark this answers is not in the document]
Comment (monopole) [reply docs/v_2.1.0.md]: thanks

File: docs/v_2.1.0.md (hunk 1 above)
Comment (monopole) [reply docs/v_2.1.0.md]: well, _customize_  is just a synonym for transform in this case, so it's shorter, but doesn't include the notion of _generating_ a generator or transformer.

**Pull request merged**